firewood/tests/db.rs

```diff
@@ -2,47 +2,28 @@
 // See the file LICENSE.md for licensing terms.
 
 use firewood::{
-    db::{BatchOp, Db as PersistedDb, DbConfig, DbError, WalConfig},
-    merkle::TrieHash,
+    db::{BatchOp, Db as PersistedDb, DbConfig, WalConfig},
+    v2::api::{self, Db, DbView, Proposal},
 };
 
-use std::{
-    collections::VecDeque,
-    fs::remove_dir_all,
-    ops::{Deref, DerefMut},
-    path::Path,
-    sync::Arc,
-};
+use std::{any::Any, collections::VecDeque, sync::Arc};
 
 // TODO: use a trait
 macro_rules! kv_dump {
     ($e: ident) => {{
         let mut s = Vec::new();
-        $e.kv_root_hash().unwrap();
-        $e.kv_dump(&mut s).unwrap();
+        let boxed_any: Box<&dyn Any> = Box::new(&$e);
+        boxed_any
+            .downcast_ref::<firewood::db::Db>()
+            .unwrap()
+            .kv_dump(&mut s)
+            .unwrap();
         String::from_utf8(s).unwrap()
     }};
 }
 
-struct Db<'a, P: AsRef<Path> + ?Sized>(PersistedDb, &'a P);
-
-impl<'a, P: AsRef<Path> + ?Sized> Db<'a, P> {
-    fn new(path: &'a P, cfg: &DbConfig) -> Result<Self, DbError> {
-        PersistedDb::new(path, cfg).map(|db| Self(db, path))
-    }
-}
-
-impl<P: AsRef<Path> + ?Sized> Drop for Db<'_, P> {
-    fn drop(&mut self) {
-        // if you're using absolute paths, you have to clean up after yourself
-        if self.1.as_ref().is_relative() {
-            remove_dir_all(self.1).expect("should be able to remove db-directory");
-        }
-    }
-}
```

Comment:
Why did you delete this? There are probably better ways to do this for sure, but if we don't have some automated way to clean up after ourselves in tests, it will inevitably get missed. It's easy enough to make the `new` function `async`.

Reply:
While I will admit that my way of doing this was not the most elegant, I think it's a bit of a regression not to have test files cleaned up. I'm all for deleting this code in favour of something better, but I would argue that that "something better" is not just stuffing everything into a temp-dir to be manually cleaned by the user (or the OS on restart).